jax/experimental/jax2tf/tests/primitive_harness.py

```diff
@@ -314,6 +314,46 @@ def parameterized(harness_group: Iterable[Harness],
   ]
 )
 
+lax_scatter = tuple(
+  # Directly from lax.scatter in tests/lax_test.py
+  Harness(
+    f"fun={f_lax.__name__}_shape={jtu.format_shape_dtype_string(shape, dtype)}_scatterindices={scatter_indices.tolist()}_updateshape={update_shape}_updatewindowdims={dimension_numbers.update_window_dims}_insertedwindowdims={dimension_numbers.inserted_window_dims}_scatterdimstooperanddims={dimension_numbers.scatter_dims_to_operand_dims}_indicesaresorted={indices_are_sorted}_uniqueindices={unique_indices}".replace(' ', ''),
+    (lambda f_lax: lambda *args: f_lax(*args[:-2], indices_are_sorted=args[-2], unique_indices=args[-1]))(f_lax),
```

Comment:
I am wondering if this can be written more simply. What is the application to `f_lax` for? Can we just write `indices_are_sorted=indices_are_sorted` and not pass `StaticArg(indices_are_sorted)`?

Reply:
The application to `f_jax` is to avoid the problem described [here](https://docs.python.org/3.4/faq/programming.html#why-do-lambdas-defined-in-a-loop-with-different-values-all-return-the-same-result). Essentially, just writing `lambda *args: f_lax(*args[:-2], indices_are_sorted=args[-2], unique_indices=args[-1])` would result in `f_lax` always being `lax.scatter_add`.

Yes, we can set `indices_are_sorted=indices_are_sorted`, but then we also have to pass `indices_are_sorted` as a parameter to the outer lambda (and same for `unique_indices`, of course).

Reply:
Do we even need a lambda? Why isn't `f_lax` enough?

Reply:
Because the functions are declared as `f(args, *, kwargs)`, the `kwargs` must be passed by name explicitly. I guess that another way to do it (that maybe is more elegant?) would be to remove the `StaticArg(indices_are_sorted)` and `StaticArg(unique_indices)` and replace this lambda by a partial application setting the `kwargs`?

Reply:
I addressed both your comments by replacing the call with a `partial` application in my latest commit. Let me know if that works :)

Reply:
This is much nicer, thanks!

I am surprised that `dimension_numbers` does not need the same treatment.

Reply:
From the definition of `scatter_add`

```python
def scatter_add(operand: Array, scatter_indices: Array, updates: Array,
                dimension_numbers: ScatterDimensionNumbers, *,
                indices_are_sorted: bool = False,
                unique_indices: bool = False) -> Array:
```

you can see that `dimension_numbers` can be passed as a positional argument, so it does not pose the same issue.